pkg/ocp/machinesets.go

```diff
@@ -89,17 +88,24 @@ func (msd *k8sMachineSetDeployer) GetWorkerNodeImage(machineSet *unstructured.Un
 		}
 
 		disks, _, _ := unstructured.NestedSlice(existing.Object, "spec", "template", "spec", "providerSpec", "value", "disks")
-		for _, o := range disks {
-			disk := o.(map[string]interface{}) // nolint:forcetypeassert // This is expected to be a map so just panic if not.
-
-			image, _, _ := unstructured.NestedString(disk, "image")
+		if len(disks) == 0 {
+			image, _, _ := unstructured.NestedString(existing.Object, "spec", "template", "spec", "providerSpec", "value", "image")
 			if image != "" {
 				return image, nil
 			}
+		} else {
+			for _, o := range disks {
+				disk := o.(map[string]interface{}) // nolint:forcetypeassert // This is expected to be a map so just panic if not.
+
+				image, _, _ := unstructured.NestedString(disk, "image")
+				if image != "" {
+					return image, nil
+				}
+			}
 		}
 	}
 
-	return "", fmt.Errorf("could not find the image of one of the worker nodes on GCP infra %q", infraID)
+	return "", fmt.Errorf("could not retrieve the image of one of the worker nodes on rhos infra %q", infraID)
```

Comment:
```suggestion
	return "", fmt.Errorf("could not retrieve the image of one of the worker nodes from the infra %q", infraID)
```

As this is used even in GCP, lets try to use some generic message.

Reply:
Done.